tests/testsuite/build_script_extra_link_arg.rs

```diff
@@ -403,3 +403,39 @@ fn build_script_extra_link_arg_examples() {
         )
         .run();
 }
+
+#[cargo_test]
+fn issue_12663() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.5.0"
+                authors = ["wycats@example.com"]
+                edition = "2015"
+
+                [lib]
+                crate-type = ["lib", "cdylib"]
+            "#,
+        )
+        .file(
+            "src/lib.rs",
+            r#"
+                #[test]
+                fn noop() {}
+            "#,
+        )
+        .file(
+            "build.rs",
+            r#"
+                fn main() {
+                    println!("cargo::rustc-cdylib-link-arg=-lhack");
```

Comment:
<https://github.com/rust-lang/cargo/pull/15326> favors `rustc-link-arg-cdylib` over this one.